### PR TITLE
Remove usage of generalize

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

We used an anonymous Report as workaround, this needs to be removed

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1201893074552404/1201481007343178/f) _(internal)_

## 🔍 What does this change?

- Removes calls to `generalize`
- Removes the `generalize` method

## 📜 Does this require a change to the docs?

The documentation for this methods were already hidden

### CODE REVIEW

File: packages/engine/bin/cli/src/main.rs [hunk 1]
```diff
@@ -73,8 +75,19 @@ pub struct SimpleExperimentArgs {
     experiment_name: ExperimentName,
 }
 
+#[derive(Debug)]
+pub struct CliError;
+
+impl fmt::Display for CliError {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt.write_str("Could not run engine")
```
Comment (Alfred-Mountfield) [inline packages/engine/bin/cli/src/main.rs:83]: Isn't this just going to result in 

```
Could not run engine

Caused by
  Could not run engine
```

File: packages/engine/lib/error/src/report.rs [hunk 2]
```diff
@@ -279,16 +279,6 @@ impl<T> Report<T> {
         )
     }
 
-    /// Converts the `Report<T>` to `Report<()>` without modifying the frame stack.
```
Comment (Alfred-Mountfield) [inline packages/engine/lib/error/src/report.rs:282]: 🎉 🎉

File: packages/engine/bin/cli/src/main.rs (hunk 1 above)
Comment (TimDiekmann) [reply packages/engine/bin/cli/src/main.rs:83]: No, it's the message which is printed when changing the context to `CliError`. Example output:
```
Error: Could not run engine
             at bin/cli/src/main.rs:133:10

Caused by:
   0: Engine didn't exit gracefully.
             at /home/ghost/Code/hash/packages/engine/lib/orchestrator/src/experiment.rs:441:9

Span trace:
   0: orchestrator::experiment::run
           with experiment_name=single_run experiment_id=bd4b4321-8e0a-4e00-b91f-4eafe99a89ae
             at lib/orchestrator/src/experiment.rs:217
```

Comment (Alfred-Mountfield) [reply packages/engine/bin/cli/src/main.rs:83]: Yeah fair point we won't have a "Caused by" as they're two different processes. The reason I ask is you've got the same error message in the CLI and also in the `hash_engine` binary. And it feels like they should be different if we're actually trying to improve our error messages

Comment (TimDiekmann) [reply packages/engine/bin/cli/src/main.rs:83]: Hmm, do you have a suggestion that makes sense for the CLI? It's almost only using the orchestrator, nothing else.

Comment (Alfred-Mountfield) [reply packages/engine/bin/cli/src/main.rs:83]: Could literally just go with "CLI encountered an error during execution"

```
Error: CLI encountered an error during execution
             at bin/cli/src/main.rs:133:10

Caused by:
   0: Engine didn't exit gracefully.
             at /home/ghost/Code/hash/packages/engine/lib/orchestrator/src/experiment.rs:441:9
```

I think we'd want to improve the message beyond just "CLI Error" if we start adding enumerations, but while it's a struct I don't think we have any other information to add

File: packages/engine/bin/hash_engine/src/main.rs [hunk 3]
```diff
@@ -8,8 +10,19 @@ use hash_engine_lib::{
     utils::init_logger,
 };
 
+#[derive(Debug)]
+pub struct EngineError;
+
+impl fmt::Display for EngineError {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt.write_str("Could not run engine")
```
Comment (Alfred-Mountfield) [inline packages/engine/bin/hash_engine/src/main.rs:18]: While we're here can we change this too. If I'm right, this error could appear at any time during the Engine's execution, whereas the phrasing here _feels_ like we're saying the Engine couldn't start.

We can copy the same phrasing as from the CLI error.

**Review by Alfred-Mountfield — APPROVED** [review]
🚀 Very pleased to get rid of those horrible temp methods :D

**Pull request merged**